src/DynamoCore/Nodes/ZeroTouch/DSFunctionBase.cs

```diff
@@ -149,9 +149,23 @@ protected override void InitializeOutputs(NodeModel model)
             // Returns a dictionary
             if (Definition.ReturnKeys != null && Definition.ReturnKeys.Any())
             {
+                var returns = Definition.Returns.ToList();
```

Comment:
Do we need ToList() here? Returns is IEnumerable and you will get count on that. Other than that LGTM.